packages/contracts/contracts/examples/UnirepVoting.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+import {Unirep} from '.././Unirep.sol';
+
+// Uncomment this line to use console.log
+// import 'hardhat/console.sol';
+import {ReputationVerifierHelper} from '../verifierHelpers/ReputationVerifierHelper.sol';
+
+contract UnirepVoting {
+    Unirep public unirep;
+    ReputationVerifierHelper public helper;
+    uint8 public numOptions;
+    uint8 public numVoters;
+    int[] public scores;
+    mapping(uint256 => uint256) public voted;
+
+    constructor(
+        Unirep _unirep,
+        ReputationVerifierHelper _helper,
+        uint8 _numOptions,
+        uint8 _numVoters,
+        uint48 _epochLength
+    ) {
+        // set unirep address
+        unirep = _unirep;
+
+        // set verifier address
+        helper = _helper;
+
+        numOptions = _numOptions;
+        numVoters = _numVoters;
+
+        scores = new int[](numOptions);
+
+        unirep.attesterSignUp(_epochLength);
+    }
+
+    // sign up users in this app
+    function userSignUp(
+        uint256[] calldata publicSignals,
+        uint256[8] calldata proof
+    ) public {
+        unirep.userSignUp(publicSignals, proof);
+    }
```

Comment:
and I think we should let the user generate epoch key proof when they sign up?
so users can receive votes (pretty much like Voteathon)
and they can join projects

Reply:
yeah this is the epoch key proof to sign up with. instead of users can join projects, i simplified by just having options to vote for (numOptions)